tests/tracking/test_rest_tracking.py

```diff
@@ -3470,40 +3472,54 @@ def test_rest_store_logs_spans_via_otel_endpoint(mlflow_client, use_async):
     if mlflow_client._store_type == "file":
         pytest.skip("FileStore does not support OTLP span logging")
 
-    experiment_id = mlflow_client.create_experiment(f"rest_store_otel_test_{use_async}")
-    root_span = mlflow_client.start_trace(
-        f"rest_store_otel_trace_{use_async}", experiment_id=experiment_id
-    )
-    otel_span = OTelReadableSpan(
-        name=f"test-rest-store-span-{use_async}",
-        context=build_otel_context(
-            trace_id=int(root_span.trace_id[3:], 16),  # Remove 'tr-' prefix and convert to int
-            span_id=0x1234567890ABCDEF,
-        ),
-        parent=None,
-        start_time=1000000000,
-        end_time=2000000000,
-        attributes={
-            SpanAttributeKey.REQUEST_ID: root_span.trace_id,
-            "test.attribute": json.dumps(f"test-value-{use_async}"),  # JSON-encoded string value
-        },
-        resource=None,
-    )
-    mlflow_span_to_log = Span(otel_span)
-
-    # Call either sync or async version based on parametrization
-    if use_async:
-        # Use asyncio.run to execute the async method
-        result_spans = asyncio.run(
-            mlflow_client._tracking_client.store.log_spans_async(
-                experiment_id=experiment_id, spans=[mlflow_span_to_log]
-            )
+    # Mock the server version check to return 3.4 if current MLflow version is < 3.4
+    # This allows the test to pass on dev MLflow versions before MLflow 3.4 is released.
+    # TODO: Remove this mock once MLflow 3.4 is released
+    if Version(mlflow.__version__) < Version("3.4"):
```

Comment:
can we use `pytest.mark.skipif` instead?

Reply:
I'd rather not do that, since we need to make sure the functionality actually works *before* launching 3.4

Reply:
Got it, thanks for the explanation. Can we apply https://github.com/mlflow/mlflow/pull/17391/files#r2299594388?